Lib/unittest/loader.py

```diff
@@ -503,15 +503,31 @@ def _makeLoader(prefix, sortUsing, suiteClass=None, testNamePatterns=None):
         loader.suiteClass = suiteClass
     return loader
 
+# Issue 5846: getTestCaseNames, makeSuite, and findTestCases are deprecated as
+#             of Python 3.10, scheduled for removal in Python 3.12.
 def getTestCaseNames(testCaseClass, prefix, sortUsing=util.three_way_cmp, testNamePatterns=None):
+    warnings.warn(
+        "getTestCaseNames() is deprecated and will be removed in Python 3.12.",
```

Comment:
"Function getTestCaseNames() is deprecated and will be removed in Python 3.12. Use the TestLoader method getTestCaseNames() instead."